src/holidayRecommendationService.js

```diff
@@ -0,0 +1,42 @@
+import { addDays, getNonWeekendHolidays } from './holidayService';
+import { getNumberOfWeekends, getPreviousWeekend } from './weekendService';
+
+export const getHolidayRecommendations = (year, vacationDays) => {
+  const holidayRecommendations = [];
+  const nonWeekendHolidays = getNonWeekendHolidays(year);
+
+  for (let i = 0; i < nonWeekendHolidays.length; i++) {
+    const holiday = nonWeekendHolidays[i];
+    const currentDate = new Date(holiday.date);
+
+    const startDate =
+      currentDate.getDay() >= 3 ? currentDate : getPreviousWeekend(currentDate);
+    const endDate = addDays(startDate, vacationDays);
+
+    const numberOfWeekends = getNumberOfWeekends(startDate, vacationDays);
+
+    let numberOfHolidays = 1;
+    while (
+      i < nonWeekendHolidays.length - 1 &&
+      new Date(nonWeekendHolidays[i + 1].date) <= endDate
+    ) {
+      i++;
+      numberOfHolidays++;
+    }
+
+    const holidayRecommendation = {
+      startDate: startDate,
+      endDate: endDate,
+      numberOfHolidays: numberOfHolidays,
+      numberOfWeekends: numberOfWeekends,
+      numberOfAnnualLeaves: vacationDays - numberOfHolidays - numberOfWeekends
```

Comment:
This could end up being a negative number depending on the desired number of vacation days.